pkg/translate/translate_value.go

```diff
@@ -330,10 +336,8 @@ name: istio-%s`
 
 	// need to do special handling for gateways and mixer
 	// ex. because deployment name should be istio-telemetry instead of istio-mixer.telemetry, we need to get rid of the prefix mixer part.
-	for _, cp := range specialComponentPath {
-		if cp == newPS && len(newP) > 2 {
-			newPS = newP[1 : len(newP)-1].String()
-		}
+	if _, exist := specialComponentPath[newPS]; exist && len(newP) > 2 {
```

Comment:
Better to use a map of [string]bool like this: https://goplay.googleplex.com/p/Kl2ngRlsnR3
Then you can write if specialComponentPath[newPS] && len(newP) > 2
Note that although it's idiomatic in Go to check for existence of map keys separately as you have, it's also ok and encouraged to use the default value being returned in the absence of a key e.g. for myMap map[string]string if myMap[key] == ""

Reply:
done